pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -225,3 +252,26 @@ func (p *ecrProvider) Provide() credentialprovider.DockerConfig {
 	}
 	return cfg
 }
+
+// getCurrentRegion queries the local EC2 metadata service and returns the name
+// of the region in which the current host is running
+func getCurrentRegion() (string, error) {
+	// Without a timeout this hangs forever. Use a really agressively low
+	// timeout because the ec2metadata service is bound to a local interface on
+	// the EC2 hosts.
+	sess, err := session.NewSession(aws.NewConfig().
+		WithHTTPClient(&http.Client{Timeout: 3 * time.Millisecond}).
+		WithMaxRetries(1))
+	if err != nil {
+		return "", err
+	}
+
+	client := ec2metadata.New(sess)
+	zone, err := client.GetMetadata("placement/availability-zone")
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
Maybe trim zone and check it isn't `""` here, just to avoid a panic on `[:len(zone)-1]` - I know it shouldn't happen, but maybe if we have an IAM metadata sanitizer in the loop

Reply:
This should not happen according to the [e2cmetadata docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) but it doesn't hurt to be paranoid. Updated.